### PR TITLE
Variable renames, code comments, and more

### PR DESCRIPTION
**Pull request by phmccarty**
I'm starting a general cleanup of diff.c and patch.c with respect to code readability and hopefully make the code base more approachable. The suffix sorting code needs similar treatment, but now that it lives in the separate source file (sufsort.c), I will save it for later.

### CODE REVIEW

File: src/diff.c [hunk 1]
```diff
@@ -607,103 +599,139 @@ int make_bsdiff_delta(char *old_filename, char *new_filename, char *delta_filena
...
+	int64_t last_offset = 0;
+	while (new_pos < new_size) {
+		// Find an exact match between old and new files, and require
+		// that more than 8 of the matching bytes "mismatch" from the
+		// previous exact match. A score (old_score) is used to track
+		// how many bytes match starting from new_pos in new, and from
+		// old_pos in the previous iteration.
+		int64_t old_score = 0;
+		int64_t new_peek;
+		for (new_peek = new_pos += match_len; new_pos < new_size; new_pos++) {
```
Comment (matthewrsj) [inline src/diff.c:616]: So new_peek gets the new_pos + match_len and new_pos gets the same thing. This is equivalent to 
```c
new_peek = new_pos = new_pos + match_len;
```
right? Just want to make sure I'm understanding this.

Comment (matthewrsj) [reply src/diff.c:616]: oh my then it increments new_pos at the end...

File: src/diff.c [hunk 2]
```diff
@@ -607,103 +599,139 @@ int make_bsdiff_delta(char *old_filename, char *new_filename, char *delta_filena
...
+				if ((new_peek + last_offset < old_size) &&
+				    (old_data[new_peek + last_offset] == new_data[new_peek])) {
+					old_score++;
 				}
 			}
 
-			if (((len == oldscore) && (len != 0)) ||
-			    (len > oldscore + 8)) {
+			if (((match_len == old_score) && (match_len != 0)) ||
+			    (match_len > old_score + 8)) {
```
Comment (matthewrsj) [inline src/diff.c:628]: The `+ 8` could use an additional comment that this is a heuristic and needs additional testing to prove that it is the right number.

File: src/diff.c [hunk 3]
```diff
@@ -607,103 +599,139 @@ int make_bsdiff_delta(char *old_filename, char *new_filename, char *delta_filena
...
 
-			lenb = 0;
-			if (scan < new_size) {
-				s = 0;
-				Sb = 0;
-				for (i = 1;
-				     (scan >= lastscan + i) && (pos >= i);
+			// Compute the length of a fuzzy match ending at the
+			// current positions in old and new files (old_pos and
+			// new_pos). At least half of the bytes match between
```
Comment (matthewrsj) [inline src/diff.c:665]: Add here that `half` is a heuristic that needs to be further tested to prove this is the best number.

**Review by matthewrsj — APPROVED** [review]
Definitely an improvement to readability and I read it closely to make sure nothing functional changed, of course testing should prove that better. +1

**Comment by phmccarty** [comment]
Thanks! Sure, I will expand on those comments to suggest that heuristics should be revisited. By the way, the magic number 8 and "50% matching bytes" come from Colin's [paper](http://www.daemonology.net/papers/bsdiff.pdf), which describes the current algorithm.

**Pull request merged**